controlplane/eks/api/v1beta2/awsmanagedcontrolplane_webhook.go

```diff
@@ -236,6 +236,10 @@ func (r *AWSManagedControlPlane) validateEKSAddons() field.ErrorList {
 		return allErrs
 	}
 
+	if r.Spec.Version == nil {
```

Comment:
@luthermonson Sorry, just one more quick question. Is this right? We stop the complete execution of the validator if the version is nil? Shouldn't this do an append to the `allErrs` list?

And when I look at the above check for addons for IPv6 I think that's also wrong. :D

Reply:
And it's actually repeated below. :D :D :D

Reply:
we can't just append it as an error because itll panic when later the func tries to dereference a nil pointer. if it's empty string or didn't parse or whatever then the LessThan checks will just fail and the error checking will trip and you'll get errors about how addons require specific k8s versions.